jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -405,6 +407,9 @@ public boolean handle()
                     {
                         try
                         {
+                            _request.getHttpInput().failed(new IOException("cancellation"));
+                            getEndPoint().cancelFillInterested();
```

Comment:
If there is an exception associated with the sendError, it should be used for the fail and the cancel.
If a new exception is needed, it should be the same exception for the failed and the cancel
perhaps httpInput.failed can do the the cancel